pkg/daemons/agent/agent.go

```diff
@@ -182,13 +184,34 @@ func checkCgroups() (kubeletRoot, runtimeRoot string, hasCFS, hasPIDs bool) {
 	}
 	defer f.Close()
 
+	v2 := cgroups.Mode() == cgroups.Unified
```

Comment:
I am not very familiar with `github.com/containerd/cgroups` behavior -- is it possible that we'll receive a `cgroups.Unified` result from `cgroups.Mode()` even if we don't have `systemd` as our init process?

i.e. on a system with unified cgroups but no systemd

Reply:
Yes,that is possible, as long as cgroup2 is mounted on /sys/fs/cgroup

Reply:
@AkihiroSuda do you know if there is a non-systemd based system that only supports cgroups v2 that we could test this on?

Reply:
OpenRC seems to support cgroup v2 (aka "unified mode") as well:

https://github.com/OpenRC/openrc/blob/170ce2624a98952e6eb871cd9915c7c1b76ab887/etc/rc.conf#L194-L199

Reply:
@brandond FYI. I built version of k3OS which combines this PR with OpenRC on "unified" mode. You can find it from: https://github.com/olljanat/k3os/releases/tag/rootless-cgroup-v2-test

Afaiu cgroup is using v2 mode and k3s still works:
```
k3os-11941 [~]$ ls -l /sys/fs/cgroup/
total 0
drwxr-xr-x 2 root root 0 Jan 25 22:04 acpid
drwxr-xr-x 2 root root 0 Jan 25 22:04 ccapply
-r--r--r-- 1 root root 0 Jan 25 22:04 cgroup.controllers
-rw-r--r-- 1 root root 0 Jan 25 22:04 cgroup.max.depth
-rw-r--r-- 1 root root 0 Jan 25 22:04 cgroup.max.descendants
-rw-r--r-- 1 root root 0 Jan 25 22:04 cgroup.procs
-r--r--r-- 1 root root 0 Jan 25 22:04 cgroup.stat
-rw-r--r-- 1 root root 0 Jan 25 22:11 cgroup.subtree_control
-rw-r--r-- 1 root root 0 Jan 25 22:04 cgroup.threads
drwxr-xr-x 2 root root 0 Jan 25 22:04 connman
-rw-r--r-- 1 root root 0 Jan 25 22:04 cpu.pressure
-r--r--r-- 1 root root 0 Jan 25 22:04 cpuset.cpus.effective
-r--r--r-- 1 root root 0 Jan 25 22:04 cpuset.mems.effective
drwxr-xr-x 2 root root 0 Jan 25 22:04 dbus
drwxr-xr-x 2 root root 0 Jan 25 22:04 haveged
-rw-r--r-- 1 root root 0 Jan 25 22:04 io.cost.model
-rw-r--r-- 1 root root 0 Jan 25 22:04 io.cost.qos
-rw-r--r-- 1 root root 0 Jan 25 22:04 io.pressure
drwxr-xr-x 2 root root 0 Jan 25 22:04 iscsid
drwxr-xr-x 2 root root 0 Jan 25 22:04 issue
drwxr-xr-x 2 root root 0 Jan 25 22:04 k3s-service
drwxr-xr-x 4 root root 0 Jan 25 22:04 kubepods
-rw-r--r-- 1 root root 0 Jan 25 22:04 memory.pressure
drwxr-xr-x 2 root root 0 Jan 25 22:04 sshd
drwxr-xr-x 2 root root 0 Jan 25 22:04 syslog
drwxr-xr-x 2 root root 0 Jan 25 22:04 udev
k3os-11941 [~]$ kubectl get nodes
NAME         STATUS   ROLES                  AGE     VERSION
k3os-11941   Ready    control-plane,master   7m41s   v1.20.2+k3s-127b96e8
```

Reply:
olljanat if you are confirming that this works with OpenRC in unified mode then I'm satisfied. We can open another issue to track hybrid mode with the cpu,pids controllers in v2.

Reply:
"hybrid" mode is almost meaningless in the container ecosystem, because a controller cannot be enabled for both v1 and v2 simultaneously.

Reply:
Yeah hybrid mostly just means we should check v2 and fall back to v1. Right now we just assume they are available in v1 in hybrid mode.

Reply:
If the case is that without this one k3s fails to start if cgroup v2 is used then I can confirm that it works OpenRC in unified mode. Or is there something else what I should check?